### PR TITLE
Retemplate the repository

### LINKED ISSUE #129: Add a codestyle check to the CI
Currently there is no style check for the CI. It would be nice to add one using something like `flake8`

### LINKED ISSUE #145: Move away from setup.py
`python setup.py install` is outdated. We should be using `pip install -e .`. I think this should mostly just involve refactoring `setup.py` and moving a bunch of information into `setup.cfg`.

### LINKED ISSUE #149: Fix links in README
There are a few places where the links in the README are stale, e.g. to the docs and to the HYDRAD source code

### LINKED ISSUE #150: Bump min Python version to 3.8

### LINKED ISSUE #153: Re-theme the docs
The current theme is kind of ugly. Some alternatives:

- [pydata theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html)
- some modified version of the [sunpy theme](https://github.com/sunpy/sunpy-sphinx-theme)

### PR DESCRIPTION
**Pull request by wtbarnes**
Fixes #145 
Fixes #150
Fixes #129 
Fixes #149 
Fixes #153

Retemplate the repository to make things a bit easier to install and package

- [x] Update install instructions
- [x] Double check that all above issues are resolved
- [x] Test only on 3.9 and above
- [x] Update CI scripts to use tox
- [x] Change "master" to "main" in CI scripts

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,41 +1,64 @@
...
-    - name: Report coverage
-      uses: codecov/codecov-action@v1
-      with:
-        fail_ci_if_error: true
-        verbose: true
-        file: ./coverage.xml
+  test:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      toxdeps: "'tox<4' tox-pypi-filter"
```
Comment (nabobalis) [inline .github/workflows/test.yml:17]: Shouldn't need this anymore.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,41 +1,65 @@
...
+    with:
+      toxdeps: tox-pypi-filter
+      envs: |
+        - macos: py39
+        - macos: py311
+        - windows: py39
+        - windows: py311
+        - linux: py39
+        - linux: py310
+        - linux: py311
```
Comment (nabobalis) [inline .github/workflows/test.yml:25]: Do you want all of these?

File: MANIFEST.in [hunk 3]
```diff
@@ -1,3 +1,5 @@
 include README.md
 include LICENSE
-include pydrad/configure/templates/*
\ No newline at end of file
+include pyproject.toml
```
Comment (nabobalis) [inline MANIFEST.in:3]: I think this file shold be empty unless you need to prune.

Between setuptools_scm and I think the data finder in setup.cfg, that should cover all of this. See https://setuptools.pypa.io/en/latest/userguide/datafiles.html

File: pydrad/configure/configure.py [hunk 4]
```diff
@@ -426,28 +424,27 @@ def _fit_poly_domains(self, name, unit):
...
         if 'minimum_cells' in self.config['grid']:
             return int(self.config['grid']['minimum_cells'])
-        n_min = self.config['general']['loop_length'] / self.config['grid']['maximum_cell_width']
+        L = self.config['general']['loop_length']
+        n_min = L / self.config['grid']['maximum_cell_width']
         if n_min.decompose().unit != u.dimensionless_unscaled:
-            raise u.UnitConversionError(
-                f'''Maximum cell width must be able to be converted to 
-                {self.config['general']['loop_length'].unit}''')
+            raise u.UnitConversionError(f'''Maximum cell width must be able to be converted to {L.unit}''')
```
Comment (nabobalis) [inline pydrad/configure/configure.py:440]: I would remove the `'''` and just have one.

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,69 @@
+[tox]
+envlist =
+    py{39,310,311}
```
Comment (nabobalis) [inline tox.ini:3]: We have 3.9 here but setup.cfg says 3.8, are you planning to drop 3.8 support?

File: .github/workflows/test.yml (hunk 2 above)
Comment (wtbarnes) [reply .github/workflows/test.yml:25]: Probably not.

File: tox.ini (hunk 5 above)
Comment (wtbarnes) [reply tox.ini:3]: Well I didn't see any reason to exclude 3.8 but also didn't want to add an additional test. I guess I should just only support 3.9+ if I'm not testing it.

File: MANIFEST.in (hunk 3 above)
Comment (wtbarnes) [reply MANIFEST.in:3]: You mean the whole manifest file should be empty?

Comment (nabobalis) [reply MANIFEST.in:3]: You should be able to delete the file.

**Pull request merged**